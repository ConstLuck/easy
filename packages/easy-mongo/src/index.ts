export * from './MongoGateway';
export * from './MongoProvider';
