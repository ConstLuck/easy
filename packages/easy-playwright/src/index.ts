export * from './PlaywrightElement';
export * from './PlaywrightTester';
