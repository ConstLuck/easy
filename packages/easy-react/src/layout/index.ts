export * from './For';
export * from './If';
