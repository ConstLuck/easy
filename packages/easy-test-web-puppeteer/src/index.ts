export * from './PuppeteerElement';
export * from './PuppeteerTester';
