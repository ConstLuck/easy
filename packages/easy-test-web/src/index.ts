export * from './Tester';
export * from './ElementTester';
