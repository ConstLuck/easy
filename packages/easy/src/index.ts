export * from './data';
export * from './domain';
export * from './http';
export * from './health';
export * from './mongo';
export * from './process';
export * from './resources';
export * from './services';
export * from './security';
export * from './sql';
export * from './types';
export * from './utils';
export * from './validation';
