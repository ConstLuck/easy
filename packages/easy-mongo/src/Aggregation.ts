import { ifDefined, isPresent, Optional } from '@thisisagile/easy';
import { Filter, FindOptions } from './MongoProvider';
import { toMongoType } from './Utils';

export const asc = 1;
export const desc = -1;

export const aggregation = {
  match: (f: Filter) => ({ $match: f }),
  id: (id: Filter) => aggregation.match({ id }),
  eq: (key: string, value: Filter) => aggregation.match({ [key]: value }),
  gt: (key: string, value: Filter) => ({ [key]: { $gt: value } }),
  gte: (key: string, value: Filter) => ({ [key]: { $gte: value } }),
  lt: (key: string, value: Filter) => ({ [key]: { $lt: value } }),
  lte: (key: string, value: Filter) => ({ [key]: { $lte: value } }),
  sum: (to: string, from: string = to) => ({ [to]: { $sum: `$${from}` } }),
  count: (to = 'count') => ({ [to]: { $count: {} } }),
  after: (key: string, date: unknown) => aggregation.match(aggregation.gt(key, toMongoType(date))),
  group: (by: Filter, ...fs: Filter[]) => ({ $group: Object.assign({ _id: by }, ...fs) }),
  skip: ({ skip: $skip }: FindOptions): Optional<Filter> => ifDefined($skip, { $skip }),
  take: ({ take: $limit }: FindOptions): Optional<Filter> => ifDefined($limit, { $limit }),
  sort: ($sort: Record<string, typeof asc | typeof desc>): Optional<Filter> => (isPresent($sort) ? { $sort } : undefined),
  asc: (key: string) => aggregation.sort({ [key]: asc }),
  desc: (key: string) => aggregation.sort({ [key]: desc }),
  date: (key: string, format = '%Y-%m-%d') => ({ $dateToString: { date: `$${key}`, format } }),
};
