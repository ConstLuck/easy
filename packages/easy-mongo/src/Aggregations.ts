import { ifDefined, isPresent, Optional } from '@thisisagile/easy';
import { Filter, FindOptions } from './MongoProvider';
import { toMongoType } from './Utils';

export const asc = 1;
export const desc = -1;

export const aggregations = {
  gt: (key: string, value: Filter) => ({ [key]: { $gt: value } }),
  gte: (key: string, value: Filter) => ({ [key]: { $gte: value } }),
  lt: (key: string, value: Filter) => ({ [key]: { $lt: value } }),
  lte: (key: string, value: Filter) => ({ [key]: { $lte: value } }),
  match: (f: Filter) => ({ $match: f }),
  id: (id: Filter) => aggregations.match({ id }),
  eq: (key: string, value: Filter) => aggregations.match({ [key]: value }),
  after: (key: string, date: unknown) => aggregations.match(aggregations.gte(key, toMongoType(date))),
  before: (key: string, date: unknown) => aggregations.match(aggregations.lt(key, toMongoType(date))),
  sum: (to: string, from: string = to) => ({ [to]: { $sum: `$${from}` } }),
  count: (to = 'count') => ({ [to]: { $count: {} } }),
  groupBy: (by: Filter) => ({
    and: (...fs: Filter[]) => ({ $group: Object.assign({ _id: by }, ...fs) }),
  }),
  skip: ({ skip: $skip }: FindOptions): Optional<Filter> => ifDefined($skip, { $skip }),
  take: ({ take: $limit }: FindOptions): Optional<Filter> => ifDefined($limit, { $limit }),
  sort: ($sort: Record<string, typeof asc | typeof desc>): Optional<Filter> => (isPresent($sort) ? { $sort } : undefined),
  asc: (key: string) => aggregations.sort({ [key]: asc }),
  desc: (key: string) => aggregations.sort({ [key]: desc }),
  date: (key: string, format = '%Y-%m-%d') => ({ $dateToString: { date: `$${key}`, format } }),
};
