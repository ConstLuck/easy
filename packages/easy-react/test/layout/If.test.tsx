import React from 'react';
import '@thisisagile/easy-test';
import { rendersWait } from '@thisisagile/easy-test-react';
import { If } from '../../src';

describe('If', () => {
  const component1 = (
    <If condition={true}>
      <div data-testid="found">Hello</div>
    </If>
  );
  const component2 = (
    <If condition={false}>
      <div data-testid="found">Hello</div>
    </If>
  );

  test('renders correctly.', async () => {
    const { container } = await rendersWait(component1);
    expect(container).toMatchSnapshot();
  });

  test('is found', async () => {
    const { atId } = await rendersWait(component1);
    expect(atId('found')).toBeValid();
  });

  test('renders correctly 2.', async () => {
    const { container } = await rendersWait(component2);
    expect(container).toMatchSnapshot();
  });

  test('is not found', async () => {
    const { atId } = await rendersWait(component2);
    expect(atId('found')).not.toBeValid();
  });
});
