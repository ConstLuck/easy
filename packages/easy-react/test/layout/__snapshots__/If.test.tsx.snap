// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If renders correctly 2. 1`] = `<div />`;

exports[`If renders correctly. 1`] = `
<div>
  <div
    data-testid="found"
  >
    Hello
  </div>
</div>
`;
